build-aux/m4/ax_pthread.m4

```diff
@@ -482,18 +487,28 @@ if test "x$ax_pthread_ok" = "xyes"; then
                     [#handle absolute path differently from PATH based program lookup
                      AS_CASE(["x$CC"],
                          [x/*],
-                         [AS_IF([AS_EXECUTABLE_P([${CC}_r])],[PTHREAD_CC="${CC}_r"])],
-                         [AC_CHECK_PROGS([PTHREAD_CC],[${CC}_r],[$CC])])])
+                         [
```

Comment:
I think a copy/paste error happened here with regard to indentation. Comparing to the source file https://raw.githubusercontent.com/autoconf-archive/autoconf-archive/master/m4/ax_pthread.m4 yields a diff here.

Reply:
Ah, they use tabs. How annoying!

Nonetheless, going to push an updated branch.

Reply:
Updated. Is it ok now?

Reply:
> Updated. Is it ok now?

No. It still differs from upstream.

```diff
diff --git a/build-aux/m4/ax_pthread.m4 b/build-aux/m4/ax_pthread.m4
index 9f35d13914..2b2e1d9a4f 100644
--- a/build-aux/m4/ax_pthread.m4
+++ b/build-aux/m4/ax_pthread.m4
@@ -488,13 +488,13 @@ if test "x$ax_pthread_ok" = "xyes"; then
                      AS_CASE(["x$CC"],
                          [x/*],
                          [
-                          AS_IF([AS_EXECUTABLE_P([${CC}_r])],[PTHREAD_CC="${CC}_r"])
-                          AS_IF([test "x${CXX}" != "x"], [AS_IF([AS_EXECUTABLE_P([${CXX}_r])],[PTHREAD_CXX="${CXX}_r"])])
-                        ],
+               AS_IF([AS_EXECUTABLE_P([${CC}_r])],[PTHREAD_CC="${CC}_r"])
+               AS_IF([test "x${CXX}" != "x"], [AS_IF([AS_EXECUTABLE_P([${CXX}_r])],[PTHREAD_CXX="${CXX}_r"])])
+             ],
                          [
-                          AC_CHECK_PROGS([PTHREAD_CC],[${CC}_r],[$CC])
-                          AS_IF([test "x${CXX}" != "x"], [AC_CHECK_PROGS([PTHREAD_CXX],[${CXX}_r],[$CXX])])
-                        ]
+               AC_CHECK_PROGS([PTHREAD_CC],[${CC}_r],[$CC])
+               AS_IF([test "x${CXX}" != "x"], [AC_CHECK_PROGS([PTHREAD_CXX],[${CXX}_r],[$CXX])])
+             ]
                      )
                     ])
                 ;;
```

Reply:
> No. It still differs from upstream.

Could it be due to tab/space settings in your environment?

In my editor the diff looks like that:

![Screenshot from 2021-12-07 14-03-02](https://user-images.githubusercontent.com/32963518/145025840-3754d6dc-728a-400a-a4ef-5bb90e1300eb.png)

Reply:
```
$ diff -u autoconf-archive/m4/ax_pthread.m4 bitcoin/build-aux/m4/ax_pthread.m4 
```
is empty.

Reply:
> Could it be due to tab/space settings in your environment?

Sorry, I must still have been comparing the previous branch. This looks ok now.

Editors and tabs-versus-spaces etc should all be irrelevant here. "Updating" this file should be as boring as:
```bash
curl https://git.savannah.gnu.org/gitweb/\?p\=autoconf-archive.git\;a\=blob_plain\;f\=m4/ax_pthread.m4 -o build-aux/m4/ax_pthread.m4
```